example/lib/customs/pickers/directory_file_asset_picker.dart

```diff
@@ -701,10 +701,12 @@ class FileAssetPickerBuilder
     int index,
     List<File> currentAssets,
   ) {
-    final int currentIndex = switch (specialItemPosition) {
-      SpecialItemPosition.none || SpecialItemPosition.append => index,
-      SpecialItemPosition.prepend => index - 1,
-    };
+    int currentIndex = index;
+
+    if (prepandSpecialItems.isNotEmpty) {
+      currentIndex = index - prepandSpecialItems.length;
+    }
```

Comment:
This could always minus the special items length.

Reply:
Agreed. Done removing redundant checking.